server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -443,7 +455,7 @@ public void moveSegment(
             () -> {
               try {
                 if (serverInventoryView.isSegmentLoadedByServer(toServer.getName(), segment) &&
-                    curator.checkExists().forPath(toLoadQueueSegPath) == null &&
+                    (curator == null || curator.checkExists().forPath(toLoadQueueSegPath) == null) &&
```

Comment:
This doesn't necessarily need to change in this PR, but it seems kind of leaky that this thing has a `CuratorFramework` at all, it seems like the load peon should provide this check so it can just be a no-op for non-zk. and then `DruidCoordinator` no longer needs a curator or zk paths I think?

Reply:
yeah, I hope zk based segment management will just go away and this will go as well.